### PR TITLE
Add React.Fragment to top-level API in React

### PR DESCRIPTION
**Pull request by uniqueiniquity**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://github.com/facebook/react/pull/10783
- [X] Increase the version number in the header if appropriate. (Not appropriate)
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. (Not substantial)

### CODE REVIEW

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

**Review by tkrotoff — CHANGES_REQUESTED** [review]

File: types/react/index.d.ts
Comment (uniqueiniquity) [reply types/react/index.d.ts]: Are you suggesting that I also add a new type alias named `Fragment` that is `symbol | number` and then use that instead as shorthand?

**Comment by RyanCavanaugh** [comment]
@uniqueiniquity Please address comments from the code reviewers.

File: types/react/index.d.ts
Comment (tkrotoff) [reply types/react/index.d.ts]: @uniqueiniquity sorry did not see your comment
> new type alias named `Fragment` that is `symbol | number`

Yes, you have also `const Fragment: symbol | number;`. It's a suggestion, your choice.

Comment (uniqueiniquity) [reply types/react/index.d.ts]: I'm a little confused by your suggestion; that's a value with the type `symbol | number`, not a type alias.

Comment (tkrotoff) [reply types/react/index.d.ts]: > that's a value [...] not a type alias

I know, you can then write `const Fragment: NEW_TYPE;`, NEW_TYPE being named something like `ReactFragment` since I guess naming it `Fragment` would clashname with the `const Fragment...`.

FYI a type `Fragment` exist in React Flow source code: https://github.com/facebook/react/blob/v16.1.0/packages/shared/ReactTypes.js#L18

Comment (uniqueiniquity) [reply types/react/index.d.ts]: Aha, I understand now. I would prefer keeping it explicit here to prevent confusion with [this type](https://github.com/uniqueiniquity/DefinitelyTyped/blob/4d07771b6c5ed5c3b75c7ee3e0d939557927403d/types/react/index.d.ts#L169).

File: types/react/test/index.ts
Comment (vsiao) [inline types/react/test/index.ts]: I'm of the opinion that the actual type of `React.Fragment` should be opaque to clients. The fact that it is a `symbol | number` is an implementation detail. As @tkrotoff suggested I would feel better about changing references to an alias like `type FragmentType = symbol & 0xeacb;`

Comment (uniqueiniquity) [reply types/react/test/index.ts]: Unfortunately, our type aliases aren't opaque. So creating something like `type FragmentType = symbol | 0xeacb;` wouldn't be visible by clients anyway.

Comment (vsiao) [reply types/react/test/index.ts]: All I'm saying is this test shouldn't be testing that `React.Fragment` is a `symbol | number`, because that's liable to change. Something like

```ts
const fragmentType: React.FragmentType = React.Fragment
```

would be appropriate.

Comment (vsiao) [reply types/react/test/index.ts]: A type like `symbol & 0xeacb` prevents users from making a mistake by passing a `number` into `createElement`.

**Pull request merged**
